internal/merger/merger.go

```diff
@@ -366,3 +366,43 @@ func toStatVarSummaryMap(in []*pbv1.VariableInfoResponse) map[string]*proto.Stat
 	}
 	return out
 }
+
+// MergeSearchStatVarResponse merges two SearchStatVarResponse.
+func MergeSearchStatVarResponse(primary, secondary *proto.SearchStatVarResponse) *proto.SearchStatVarResponse {
+	mergedStatVars := []*proto.EntityInfo{}
+	dedupedMatches := []string{}
+	matchesMap  := map[string]bool{}
+
+	if primary != nil {
+		mergedStatVars = append(mergedStatVars, primary.StatVars...)
+
+		for _, m := range primary.Matches {
+			if _, ok := matchesMap[m]; ok {
```

Comment:
Can check for `!ok` instead and eliminate the `else` block.

Reply:
Done.